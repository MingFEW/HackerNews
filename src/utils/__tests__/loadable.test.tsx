import { render } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import { lazyLoad } from '../loadable'

const LazyComponenWithDefaultExport = lazyLoad(() => import('../../../internals/testing/loadable.mock'))

describe('loadable', () => {
  it('should render LazyComponent after waiting for it to load', async () => {
    let comp: any
    await act(() => {
      comp = render(<LazyComponenWithDefaultExport />)
    })
    expect(comp.container.firstChild).toMatchSnapshot()
  })
})
