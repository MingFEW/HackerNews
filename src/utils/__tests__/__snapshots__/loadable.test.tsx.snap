// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadable should render LazyComponent after waiting for it to load 1`] = `
<div>
  My lazy-loaded component
</div>
`;
